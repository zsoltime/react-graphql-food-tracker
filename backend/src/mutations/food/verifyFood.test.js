'use strict';

const verifyFood = require('./verifyFood');

describe('verifyFood mutation', () => {
  it("updates food's verification status", async () => {
    const args = { id: 'validFoodId' };
    const mockedContext = {
      request: { userId: 'validUserId' },
      db: {
        mutation: {
          updateFood: jest.fn().mockImplementation(() => Promise.resolve({
            id: args.id,
            name: 'Fake food',
            verifiedAt: '2018-11-27T01:19:40.270Z',
          })),
        },
        query: {
          user: jest
            .fn()
            .mockImplementation(() => Promise.resolve({ permissions: ['ADMIN'] })),
        },
      },
    };

    const food = await verifyFood(null, args, mockedContext);

    expect(mockedContext.db.mutation.updateFood).toHaveBeenCalled();
    expect(
      mockedContext.db.mutation.updateFood.mock.calls[0][0]
    ).toEqual(
      expect.objectContaining({
        data: { verifiedAt: expect.any(Date) },
        where: { id: args.id },
      })
    );
    expect(food).toMatchSnapshot();
  });
  it('throws error if user has no permission', () => {
    const args = { id: 'validFoodId' };
    const mockedContext = {
      request: { userId: 'validUserId' },
      db: {
        query: {
          user: jest
            .fn()
            .mockImplementation(() => Promise.resolve({ permissions: ['USER'] })),
        },
      },
    };

    expect(
      verifyFood(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
