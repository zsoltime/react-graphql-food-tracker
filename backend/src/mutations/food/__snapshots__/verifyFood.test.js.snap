// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verifyFood mutation throws error if user has no permission 1`] = `"You don't have permission to verify foods"`;

exports[`verifyFood mutation updates food's verification status 1`] = `
Object {
  "id": "validFoodId",
  "name": "Fake food",
  "verifiedAt": "2018-11-27T01:19:40.270Z",
}
`;
